src/service/deletion.py

```diff
@@ -102,71 +207,81 @@ async def _delete_matches(deps: PickleableDependencies):
     try:
         specs = {s.data_product: s for s in data_product_specs.get_data_products()}
         async def proc(m):
-            await _delete_match(storage, specs, m)
+            await _delete_subset(storage, specs, m, models.SubsetType.MATCH)
         await storage.process_deleted_matches(proc)
     finally:
         await cli.close()
 
 
-class MatchCleanup:
+async def _delete_selections(deps: PickleableDependencies):
+    logging.basicConfig(level=logging.INFO)
+    _logger().info("Starting selection data deletion process")
+    cli, storage = await deps.get_storage()
```

Comment:
I am not sure if storage needs to be managed and released after. 
something like:
`async with deps.get_storage() as storage:
        cli, _ = storage`